equinix/resource_metal_device_acc_test.go

```diff
@@ -1060,6 +1061,16 @@ func mockFuncNotImplemented(f string) error {
 
 var _ packngo.DeviceService = (*mockDeviceService)(nil)
 
+type mockDevicesApi struct {
```

Comment:
Not sure exactly where I left this conversion, but I think it'd be better to update the tests to use a mocked API rather than mock whatever SDK is currently in use.

That should be doable with `httptest`, based on this: https://medium.com/zus-health/mocking-outbound-http-requests-in-go-youre-probably-doing-it-wrong-60373a38d2aa#fb9e

Based on that post I think we'd need to set up a mock API with `httptest`, configure the test server to handle the expected API requests, and include a `provider` block in the test config that points the provider at the `httptest` server.

Reply:
See #371 for the `httptest` change